src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -1203,16 +1203,19 @@ public void NotifyAboutClusterTopologyAndConnectivityChanges()
 
         private void OnTopologyChangeInternal(ClusterTopology topology, Dictionary<string, NodeStatus> status = null)
         {
-            if (_lastClusterTopologyIndex < topology.Etag)
```

Comment:
I noticed we reach this function even if there was a change unrelated to topology so I moved the notification and topology update into this check

Reply:
Though the url check inside `ShouldUpdateTopology` is also checking that the node we are on exists in the current cluster topology. I wonder if this could cause problems.. (in the case of removing a node from cluster maybe?)

Reply:
Don't think so, because you are the one that is not in the topology so... the DocumentDatabase will be disposed?